resources/boolean.rb

```diff
@@ -16,18 +16,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-selinux_install 'selinux os prep'
+unified_mode true
 
-selinux_state "SELinux #{node['selinux']['state'].capitalize}" do
-  action node['selinux']['state'].downcase.to_sym
-end
+property :boolean, String,
+          required: true
+
+property :value, [Integer, String, TrueClass, FalseClass],
+          required: true
 
-node['selinux']['booleans'].each do |boolean, value|
-  value = SELinuxServiceHelpers.selinux_bool(value)
-  next if value.nil?
-  script "boolean_#{boolean}" do
+action :set do
+  script "boolean_#{new_resource.boolean}" do
```

Comment:
It's probably outside the scope of this, but the ideal setup here is to load the state of this and converge only if you need to. That way you get reporting showing the current state and what was converged to

Reply:
I've done it in other cookbooks, shouldn't be a problem and makes sense.